haystack/modeling/model/language_model.py

```diff
@@ -835,840 +390,570 @@ def load(cls, pretrained_model_name_or_path: Union[Path, str], language: str = N
-        passage_attention_mask = passage_attention_mask.view(-1, max_seq_len)
         output_tuple = self.model(
-            input_ids=passage_input_ids,
-            token_type_ids=passage_segment_ids,
-            attention_mask=passage_attention_mask,
-            return_dict=True,
+            input_ids=input_ids, token_type_ids=segment_ids, attention_mask=attention_mask, return_dict=return_dict
         )
-        if self.model.ctx_encoder.config.output_hidden_states == True:
+        if output_hidden_states or self.encoder.config.output_hidden_states:
             pooled_output, all_hidden_states = output_tuple.pooler_output, output_tuple.hidden_states
             return pooled_output, all_hidden_states
-        else:
-            pooled_output = output_tuple.pooler_output
-            return pooled_output, None
-
-    def enable_hidden_states_output(self):
-        self.model.ctx_encoder.config.output_hidden_states = True
-
-    def disable_hidden_states_output(self):
-        self.model.ctx_encoder.config.output_hidden_states = False
-
 
-class BigBird(LanguageModel):
+        pooled_output = output_tuple.pooler_output
+        return pooled_output, None
+
+        # # If inheriting from HFLanguageModel
+        # output_tuple = super().forward(
+        #     input_ids=input_ids,
+        #     segment_ids=segment_ids,
+        #     attention_mask=attention_mask,
+        #     output_hidden_states=output_hidden_states,
+        #     output_attentions=output_attentions,
+        #     return_dict=return_dict
+        # )
+        # if output_hidden_states or self.encoder.config.output_hidden_states:
+        #     pooled_output, all_hidden_states = output_tuple.pooler_output, output_tuple.hidden_states
+        #     return pooled_output, all_hidden_states
+
+        # pooled_output = output_tuple.pooler_output
+        # return pooled_output, None
+
+
+#: Match the name of the HuggingFace Model class to the corresponding Haystack wrapper
+HUGGINGFACE_TO_HAYSTACK: Dict[str, Union[Type[HFLanguageModel], Type[DPREncoder]]] = {
+    "Auto": HFLanguageModel,
+    "Albert": HFLanguageModel,
+    "Bert": HFLanguageModel,
+    "BigBird": HFLanguageModel,
+    "Camembert": HFLanguageModel,
+    "Codebert": HFLanguageModel,
+    "DebertaV2": HFLanguageModelWithPooler,
+    "DistilBert": HFLanguageModelNoSegmentIds,
+    "DPRContextEncoder": DPREncoder,
+    "DPRQuestionEncoder": DPREncoder,
+    "Electra": HFLanguageModelWithPooler,
+    "GloVe": HFLanguageModel,
+    "MiniLM": HFLanguageModel,
+    "Roberta": HFLanguageModel,
+    "Umberto": HFLanguageModel,
+    "Word2Vec": HFLanguageModel,
+    "WordEmbedding_LM": HFLanguageModel,
+    "XLMRoberta": HFLanguageModel,
+    "XLNet": HFLanguageModelWithPooler,
+}
+#: HF Capitalization pairs
+HUGGINGFACE_CAPITALIZE = {
+    "xlm-roberta": "XLMRoberta",
+    "deberta-v2": "DebertaV2",
+    **{k.lower(): k for k in HUGGINGFACE_TO_HAYSTACK.keys()},
+}
+
+
+def capitalize_and_get_class(
```

Comment:
@ZanSara I know it's an instinctual habit to name a method like this with `and` but that's a smell IIRC. So instead of `capitalize_and_get_class` why not `resolve_lm_class` or something similar? Also, the model_type gets mysteriously changed/capitalized as the outcome of this method which is hard to trace IMHO. Thoughts?

Reply:
I really don't like this either, I'm glad you spotted it. 

The issue is that very often `model_type` is lower case, while to retrieve the correct class from `transformers` I need the capitalized version of it. I could do two separate functions of course, but then I repeatedly fell into the trap of "Let's not capitalize here, the value is already capitalized" just to find that one test that failed because, through another code path, I was receiving a lower case version again. 

However, now I realized that at this point the code should be clean enough and I can replace all calls to `capitalize_and_get_class` with two separate calls. I will have a look and split this.